app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -2,30 +2,58 @@ package com.mapbox.vision.teaser
 
 import android.content.Intent
 import android.view.View
-import android.widget.ImageView
 import android.widget.LinearLayout
+import android.widget.Toast
 import com.mapbox.vision.VisionManager
+import com.mapbox.vision.VisionReplayManager
+import com.mapbox.vision.mobile.core.models.FrameStatistics
 import com.mapbox.vision.teaser.view.BaseTeaserActivity
 import com.mapbox.vision.teaser.view.show
 import com.mapbox.vision.safety.VisionSafetyManager
+import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.*
 import com.mapbox.vision.teaser.ar.ArMapActivity
+import com.mapbox.vision.teaser.replayer.ArReplayNavigationActivity
+import com.mapbox.vision.teaser.replayer.ReplayModeFragment
+import com.mapbox.vision.teaser.view.hide
 import com.mapbox.vision.view.VisionView
+import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : BaseTeaserActivity() {
+class MainActivity : BaseTeaserActivity(), ReplayModeFragment.OnClickModeItemListener {
+
+    enum class VisionManagerMode {
+        Realtime,
+        Replay,
+        Recorder
+    }
+
+    private var mode = Realtime
+    private var sessionPath = ""
 
     override fun initViews(root: View) {
         root.findViewById<LinearLayout>(R.id.ar_navigation_button_container).apply {
-            setOnClickListener { startActivity(Intent(this@MainActivity, ArMapActivity::class.java)) }
+            setOnClickListener {
+                when (mode) {
+                    Realtime -> startActivity(Intent(this@MainActivity, ArMapActivity::class.java))
+                    Replay -> startArSession()
+                    else -> return@setOnClickListener
+                }
+            }
         }
 
-        root.findViewById<ImageView>(R.id.title_teaser).apply {
-            show()
+        root.findViewById<LinearLayout>(R.id.replay_mode_button_container).apply {
+            setOnClickListener { showReplayModeFragment() }
         }
     }
 
-    override fun getFrameStatistics() = VisionManager.getFrameStatistics()
-
     override fun initVisionManager(visionView: VisionView): Boolean {
+        return when (mode) {
+            Realtime -> initVisionManagerRealtime(visionView)
+            Replay -> initVisionManagerReplay(visionView, sessionPath)
+            Recorder -> false
+        }
+    }
+
+    private fun initVisionManagerRealtime(visionView: VisionView): Boolean {
```

Comment:
IMHO not that clear what `realtime` stands for.
I would stick to `initVisionManager` and `initReplayVisionManager` methods.

Reply:
good notice, I've removed realtime naming in next branches and keep the only Camera and Replay modes and naming.

Reply:
This two modes (Camera & Replay) fits fine 3 scenarios with Camera\Replay\Recording,